File: packages/plugin-simplicity/fixtures/ts/accessors.ts
interface Entry {
  name: string
  size: number
}

// The pair `no-duplicated-helper` must not report: one shape, two questions. A property is
// not a name the function declares, so `.name` and `.size` stay in the fingerprint, and they
// are what tell the two apart.
export function readName(entry: Entry): string {
  return entry.name
}

export function readSize(entry: Entry): number {
  return entry.size
}

File: packages/plugin-simplicity/fixtures/rust/accessors.rs
pub struct Entry {
    pub name: String,
    pub size: u64,
}

// The pair `no-duplicated-helper` must not report: one shape, two questions. Rust
// spells a field access with `field_identifier`, so `name` and `size` are never renamed
// away and the two fingerprints stay apart.
pub fn read_name(entry: &Entry) -> String {
    entry.name.clone()
}

pub fn read_size(entry: &Entry) -> u64 {
    entry.size.clone()
}
